src/game/logic/object/update/physicsupdate.cpp

```diff
@@ -648,3 +912,504 @@ Module *PhysicsBehavior::Friend_New_Module_Instance(Thing *thing, ModuleData con
 {
     return NEW_POOL_OBJ(PhysicsBehavior, thing, module_data);
 }
+
+bool Perps_Logically_Equal(float f1, float f2)
+{
+    return GameMath::Fabs(f1 - f2) <= 0.15f;
+}
+
+bool PhysicsBehavior::Check_For_Overlap_Collision(Object *other)
+{
+    if (Is_Low_Velocity(Get_Velocity())) {
+        return false;
+    }
+
+    Object *obj = Get_Object();
+    bool squish1 = obj->Can_Crush_Or_Squish(other, TEST_TYPE_0);
+    bool squish2 = other->Can_Crush_Or_Squish(obj, TEST_TYPE_0);
+
+    if (squish1 && squish2) {
+        captainslog_dbgassert(false,
+            "%s (Crusher:%d, Crushable:%d) is attempting to crush %s (Crusher:%d, Crushable:%d) but it is reciprocating -- "
+            "sh"
+            "ouldn't be possible!",
+            obj->Get_Template()->Get_Name().Str(),
+            obj->Get_Crusher_Level(),
+            obj->Get_Crushable_Level(),
+            other->Get_Template()->Get_Name().Str(),
+            other->Get_Crusher_Level(),
+            other->Get_Crushable_Level());
+        return false;
+    }
+
+    if (squish2) {
+        return true;
+    } else if (!this) {
```

Comment:
`!this` ? This can not be right, no? It would mean `this` is null. But `this` cannot be null.

Reply:
That's what the original binary is doing. Who knows why it's doing it.